types/aws-lambda/trigger/codebuild-cloudwatch-state.d.ts

```diff
@@ -0,0 +1,110 @@
+export type CodeBuildSourceLocationType =
+    | 'CODECOMMIT'
+    | 'CODEPIPELINE'
+    | 'GITHUB'
+    | 'GITHUB_ENTERPRISE'
+    | 'BITBUCKET'
+    | 'S3'
+    | 'NO_SOURCE';
+export type CodeBuildEnvironmentType =
+    | 'LINUX_CONTAINER'
+    | 'LINUX_GPU_CONTAINER'
+    | 'WINDOWS_CONTAINER'
+    | 'ARM_CONTAINER';
+export type CodeBuildEnvironmentPullCredentialsType = 'CODEBUILD' | 'SERVICE_ROLE';
+export type CodeBuildEnvironmentComputeType =
+    | 'BUILD_GENERAL1_SMALL'
+    | 'BUILD_GENERAL1_MEDIUM'
+    | 'BUILD_GENERAL1_LARGE'
+    | 'BUILD_GENERAL1_2XLARGE';
+export type CodeBuildEnvironmentVariableType = 'PARAMETER_STORE' | 'PLAINTEXT';
+
+export interface CodeBuildCloudWatchStateEvent {
+    version: string;
+    id: string;
+    'detail-type': 'CodeBuild Build State Change';
+    source: 'aws.codebuild';
+    account: string;
+    time: string;
+    region: string;
+    resources: string[];
+    detail: {
+        'build-status': CodeBuildStateType;
+        'project-name': string;
+        'build-id': string;
+        'current-phase': CodeBuildPhaseType;
+        'current-phase-context': string;
+        version: string;
+        'additional-information': {
+            cache: {
+                type: CodeBuildCacheType;
+            };
+            'build-number': number;
+            'timeout-in-minutes': number;
+            'build-complete': boolean;
+            initiator: string;
+            'build-start-time': string;
+            source: {
+                buildspec: string;
+                location: string;
+                type: CodeBuildSourceLocationType;
+            };
+            'source-version': string;
+            artifact: {
+                location: string;
+            };
+            environment: {
+                image: string;
+                'privileged-mode': boolean;
+                'image-pull-credentials-type'?: CodeBuildEnvironmentPullCredentialsType;
+                'compute-type': CodeBuildEnvironmentComputeType;
+                type: CodeBuildEnvironmentType;
+                'environment-variables': Array<{
+                    name: string;
+                    type?: CodeBuildEnvironmentVariableType;
+                    value: string;
+                }>;
+            };
+            'project-file-system-locations': [];
+            logs: {
+                'group-name': string;
+                'stream-name': string;
+                'deep-link': string;
+            };
+            phases: Array<{
+                'phase-context'?: string[]; // Not available for COMPLETED phase-type
+                'start-time': string;
+                'end-time'?: string; // Not available for COMPLETED phase-type
+                'duration-in-seconds'?: number; // Not available for COMPLETED phase-type
+                'phase-type': CodeBuildPhaseType;
+                'phase-status'?: CodeBuildPhaseStatusType; // Not available for COMPLETED phase-type
```

Comment:
I suppose you could use phase-type as a union descriminator instead of these being optional, but it probably makes it more annoying to use overall: you would know better than me!

Reply:
I think it would make stuff a bit more complicated in the end. However if you feel strong about it I can modify it

Reply:
Not at all!